src/Lykke.Sdk/LykkeServiceCollectionContainerBuilderExtensions.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Reflection;
+using Autofac;
+using Autofac.Extensions.DependencyInjection;
+using Common.Log;
+using JetBrains.Annotations;
+using Lykke.Common.ApiLibrary.Swagger;
+using Lykke.Sdk.Settings;
+using Lykke.SettingsReader;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Lykke.Sdk
+{
+    [PublicAPI]
+    public static class LykkeServiceCollectionContainerBuilderExtensions
+    {
+        public static IServiceProvider BuildServiceProvider<TAppSettings>(this IServiceCollection services, Action<LykkeServiceOptions> serviceOptionsBuilder)
+            where TAppSettings : BaseAppSettings
+        {
+            var serviceOptions = new LykkeServiceOptions();
+            serviceOptionsBuilder(serviceOptions);
+
+            if (string.IsNullOrWhiteSpace(serviceOptions.ApiVersion))
+                throw new ArgumentException("Api version must be provided.");
+
+            if (string.IsNullOrWhiteSpace(serviceOptions.ApiTitle))
+                throw new ArgumentException("Api title must be provided.");
+
+            if (serviceOptions.LogsConnectionStringFactory == null)
+                throw new ArgumentException("Logs connection string factory must be provided.");
+
+            services.AddMvc()
+                .AddJsonOptions(options =>
+                {
+                    options.SerializerSettings.ContractResolver =
+                        new Newtonsoft.Json.Serialization.DefaultContractResolver();
+                });
+
+            services.AddSwaggerGen(options =>
+            {
+                options.DefaultLykkeConfiguration(serviceOptions.ApiVersion, serviceOptions.ApiTitle);
+            });
+
+            var configurationRoot = new ConfigurationBuilder().AddEnvironmentVariables().Build();
+            var settings = configurationRoot.LoadSettings<TAppSettings>();
+
+            var builder = new ContainerBuilder();
+
+            builder.RegisterInstance(configurationRoot).As<IConfigurationRoot>();
+            builder.RegisterInstance(settings);
+            builder.RegisterInstance(serviceOptions);
+            builder.RegisterModule(new SdkModule<TAppSettings>(settings, serviceOptions.LogsConnectionStringFactory, serviceOptions.LogsTableName));
+            builder.RegisterAssemblyModules(Assembly.GetCallingAssembly());
```

Comment:
It will not work. Almost all Modules even SdkModule requires input arguments, settings as an example. Here we force them work without arguments.

Reply:
See implementation of SdkModule registration. Other modules should not have any parameters. Settings must be resolved via factory.

Reply:
Right. They should not, but the most of them have. If we want to use the Sdk also with existed services we have to do a lot of work for changing modules. It is expensive and errors prone.

Reply:
Modules with parameters wont work with Sdk, it's not an option and it's won't be an option. Yes, need all that stuff in each service. It's not so expensive actually.